app/src/components/ArtistPage.jsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import ArtistEvents from "./ArtistEvents";
+
+
+
+export default function ArtistPage() {
+
+    const { id } = useParams()
+    console.log(id);
+    const [artistId, setArtistId] = useState("");
+    const [name, setName] = useState("");
+    const [image, setImage] = useState("");
+    const [bio, setBio] = useState("");
+
+    useEffect(() => {
+        async function load() {
+            let rawResponse = await fetch(`/data/artists/${id}`)
+            if (rawResponse.ok) {
+                let response = await rawResponse.json();
+                setArtistId(response.id)
+                setName(response.name)
+                setImage(response.image)
+                setBio(response.bio)
+            }
+        }
+        load()
+    }, [id])
+
+    // If the fetching of the artist's name etc is not done yet, do not render anything..
+    if (!name) return <></>;
```

Comment:
might want to see if id fits better here, or a separate state entirely akin to "isLoaded" or something